EssentialFeed/EssentialFeedTests/Image Comments/API/ImageCommentsMapperTests.swift

```diff
@@ -0,0 +1,79 @@
+//
+// Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import XCTest
+@testable import EssentialFeed
+
+class ImageCommentsMapperTests: XCTestCase {
+	func test_map_throwsErrorOnNon2xxHTTPResponse() throws {
+		let json = makeItemsJSON([])
+		let samples = [199, 300, 400, 500]
+
+		try samples.forEach { code in
+			XCTAssertThrowsError(
+				try ImageCommentsMapper.map(json, from: HTTPURLResponse(statusCode: code))
+			)
+		}
+	}
+
+	func test_map_throwsErrorOn2xxHTTPResponseWithInvalidJSON() throws {
+		let invalidJSON = Data("invalid json".utf8)
+		let samples = [200, 201, 202, 203]
```

Comment:
Since we expect the `mapper` to accept `any2xx` statusCode, we should ensure that behavior by testing against multiple `2xx` values, including the boundaries, to test for off-by-one errors.
```suggestion
		let samples = [200, 201, 202, 203, 299]
```